modules/mobile/app/services/mobile/v2/appointments/provider_name_aggregator.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
```

Comment:
Keeping this logic in a service object because the proxy was getting very busy and because the specs for this are growing harder to write as we add more external calls, which is making the VCR cassettes difficult to work with. Extracting into a service object makes it easier to write extensive unit tests that cover edge cases. The hard part was determining where it should go and what it should be called. I decided to leave it in the folder alongside the proxy because it's pretty specific to the proxy. And, despite conversations I've had recently with @Tonksthebear about using "model" like names, I had to go with an action name here because nothing about this feels like a model to me.